lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -0,0 +1,103 @@
+"""
+Content Library Transformer, used to filter course structure per user.
+"""
+import json
+from courseware.access import _has_access_to_course
+from courseware.models import StudentModule
+from opaque_keys.edx.locator import BlockUsageLocator
+from openedx.core.lib.block_cache.transformer import BlockStructureTransformer
+
+
+class ContentLibraryTransformer(BlockStructureTransformer):
+    """
+    Content Library Transformer Class
+    """
+    VERSION = 1
+
+    @classmethod
+    def _get_selected_modules(cls, user, course_key, block_key):
+        """
+        Get list of selected modules in a library,
+        for user.
+
+        Arguments:
+            user (User)
+            course_key (CourseLocator)
+            block_key (BlockUsageLocator)
+
+        Returns:
+            list[modules]
+        """
+        return StudentModule.objects.filter(
```

Comment:
@dino-cikatic This won't work if the blocks haven't already been selected for the user.